nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -96,6 +97,12 @@ addEntry() {
     fi
     echo "  LINUX ../nixos/$(basename $kernel)"
     echo "  INITRD ../nixos/$(basename $initrd)"
+    echo "  APPEND init=$path/init $extraParams"
```

Comment:
Change seems fine. Do we have a test for extlinux-conf-builder?

Reply:
I don't think so. Perhaps we can add an "SD card image" for x86 that enables `boot.loader.generic-extlinux-compatible` and add a test to `boot.nix` with U-Boot as the BIOS? I don't much experience with the tests, though.

Reply:
I think we should. I am a bit paranoid when it comes to bootloader generator scripts. It is critical code since messing up them mean one can no longer rollback. This is the first time I am reviewing the extlinux-conf-builder.
What I am missing also right now is a file system sync at the very end of the boot directory.
This is crucial when the filesystem has no journal i.e. vfat. Some firmware might reject a fat32 filesystem with the dirty bit set hence the filesystem should be synched afterwards to leave the system for a shorter period of time in a critical state.

Reply:
I would even fsync `tmpFile` before overriding the original.

Reply:
Sure, I'll try to add a simple test in a bit.

Reply:
Some hints about the QEMU U-Boot builds here in the README:

 - https://github.com/Tow-Boot/Tow-Boot/tree/development/boards/uBoot

Reply:
Cool, thanks. I have already added a new qemu package, and I'm waiting for the image to build.

Reply:
Hmm, U-Boot somehow doesn't seem to want to boot our kernel image. The system resets due to triple fault shortly after bzImage is loaded with two general protection faults, which seems to be the behavior in `__reset_cpu` (normally it should print a message).

Reply:
Anyways I pushed my changes so maybe you can take a look. The `ubootExtlinux` test will simply be stuck in a bootloop, so you may need to run qemu manually to test it.